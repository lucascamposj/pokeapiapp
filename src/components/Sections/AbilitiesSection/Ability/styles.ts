import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  height: 60px;
  padding: 0;
  margin: 0;
  justify-content: center;
  align-items: center;
  flex-direction: row;
`;

export const AbilitiesContent = styled.View`
  flex-direction: column;
`;

export const AbilityText = styled.Text`
  text-transform: capitalize;
  font-family: 'RobotoMono-Regular';
`;

export const AbilityNumber = styled.Text`
  text-transform: uppercase;
  font-family: 'RobotoMono-Light';
`;
