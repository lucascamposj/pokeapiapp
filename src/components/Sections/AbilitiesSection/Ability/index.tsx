import React from 'react';
import { Container, AbilityText } from './styles';
import { AbilityInterface } from '../../../../pages/Pokemon';

const Ability: React.FC<AbilityInterface> = ({ ability }: AbilityInterface) => {
  return (
    <Container>
      <AbilityText>{ability.name}</AbilityText>
    </Container>
  );
};

export default Ability;
