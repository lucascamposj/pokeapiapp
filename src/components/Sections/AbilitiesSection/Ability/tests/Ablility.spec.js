import React from 'react';
import {shallow} from 'enzyme';
import Ability from '../../Ability';

describe('Ability', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const ability = [
              {
                ability: {
                  name: "Overgrow"
                }
              },
              {
                ability: {
                  name: "Chlorophyll"
                }
              }
            ]
            const component = shallow(
              <Ability
                ability={ability}
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
