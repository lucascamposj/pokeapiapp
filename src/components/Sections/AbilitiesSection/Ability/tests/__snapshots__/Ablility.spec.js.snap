// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ability Rendering should match to snapshot 1`] = `
<Styled(View)>
  <Styled(Text) />
</Styled(View)>
`;
